scipy/optimize/_basinhopping.py

```diff
@@ -21,7 +27,9 @@ def _add(self, minres):
         self.minres.x = np.copy(minres.x)
 
     def update(self, minres):
-        if minres.fun < self.minres.fun:
+        cond1 = minres.fun < self.minres.fun and minres.success
+        cond2 = minres.success and not self.minres.success
+        if cond1 or cond2:
```

Comment:
Alternatively:
```suggestion
        if minres.success and (minres.fun < self.minres.fun
                               or not self.minres.success)
```
The incumbent (the best result found so far) should be updated if the new result is comes with `success=True` and either:
- its function value is lower than that of the incumbent or
- the incumbent has `success=False`. (This latter can happen if the initial minimization was unsuccessful.)

Reply:
I like the suggestion better.

Reply:
Sure, I'll commit when this is ready for a final round of CI.